catchup/service.go

```diff
@@ -307,7 +307,8 @@ func (s *Service) fetchAndWrite(r basics.Round, prevFetchCompleteChan chan bool,
 				}
 
 				if s.cfg.CatchupVerifyTransactionSignatures() || s.cfg.CatchupVerifyApplyData() {
-					vb, err := s.ledger.Validate(s.ctx, *block, s.blockValidationPool)
+					var vb *ledger.ValidatedBlock
+					vb, err = s.ledger.Validate(s.ctx, *block, s.blockValidationPool)
```

Comment:
@tsachiherman this is the most significant issue I found — before this change, the value of `err = s.ledger.AddValidatedBlock(*vb, *cert)` on line 319/320 is ineffectual, and has no impact later on line 324/325 where it checks `if err != nil`